packages/design-system-tokens/src/lib/utility.ts

```diff
@@ -1,32 +1,105 @@
 import fs from 'fs';
 import path from 'path';
-import { HexValue, RGBValue, FileDescriptor, PercentageValue } from './types';
+import { HexValue, RGBValue, RGBAValue, FileDescriptor } from './types';
 
 // converts an rgb string 'rgb(15,24,128)' to a hex value '#0819A9'
 export const rgbToHex = (r: number, g: number, b: number): HexValue => {
   const hex = ((1 << 24) + (r << 16) + (g << 8) + b).toString(16).slice(1);
   return `#${hex}`;
 };
 
+// transforms a hex value to an 8 char hex value with opacity given as a number
+export const hexOpacity = (hexVal: HexValue, opacity: number): HexValue => {
+  const percent = Math.max(0, Math.min(100, opacity));
+  const intVal = Math.round((percent / 100) * 255);
+  const hexOpacity = intVal.toString(16).toUpperCase().padStart(2, '0');
+  return `${hexVal}${hexOpacity}`;
+};
+
 /*
- * transforms a hex value to an 8 char hex value with opacity given as a number from 0 to 1
+ * converts a hex string '#F3G1AA' to an rgb value string 'rgb(142, 24, 89)'
+ * if an optional opacity value is passed the rgba value will be returned
  */
-export const hexOpacity = (hexVal: HexValue, opacity: number): HexValue => {
-  const alpha = Math.round(opacity * 255);
-  const alphaHex = (alpha + 0x10000).toString(16).substr(-2).toUpperCase();
-  return `${hexVal}${alphaHex}`;
+export const hexToRgbString = (hex: HexValue, opacity = 100): RGBValue | RGBAValue => {
```

Comment:
Is this used anywhere?

Reply:
Not at the moment but I can see it maybe having use if a subsystem requires rgb/rgba values down the line.. I've tested and confirmed it works correctly so just a quality of life item.